Enhance the getting started guide: scheduled backups (#1746)
vitessio/website: content/en/docs/20.0/user-guides/operating-vitess/backup-and-restore/scheduled-backups.md

---
title: Scheduled Backups
weight: 5
aliases: ['/docs/user-guides/backup-and-restore/']
---

{{< info >}}
This guide follows on from the Get Started guides. Please make sure that you have

> an [Operator](../../../../get-started/operator) or [local](../../../../get-started/local) installation ready. It also assumes

Reviewer: Since this works only with operator, we should remove the reference to local installation from here.

Author: Fixed via 5a977a20

that the [MoveTables](../../../migration/move-tables/) and [Resharding](../../../configuration-advanced/resharding) user guides have been followed (which take you through
steps `101` to `306`).

> This step is doable only if you are using the vitess-operator.

Reviewer: ```suggestion
This guide is useful only if you are using the vitess-operator.
```

{{< /info >}}

## Backups

> If you are not familiar with how backups in Vitess work we suggest you familiarize yourself with it.

Reviewer: ```suggestion
If you are not already familiar with how backups work in Vitess we suggest you familiarize yourself with them first.
```

> An [entire guide](../overview/) is available in this section.

Reviewer: Remove this line, and hyperlink from [how backups work]

Author: Fixed via 5a977a20

## Scheduling backups

### Adding the schedule

For this example we are going to create two schedules: each will be executed every minute, the first one will backup
the two `customer` shards, and the second one will backup the `commerce` keyspace.

> The backups will be stored directly inside the Minikube node, but if you want to backup to for instance S3, you can

Reviewer: ```suggestion
The backups will be stored directly inside the Minikube node, but if you want to backup to a cloud storage provider like S3, you can
```

change the `location` of the backup in `401_scheduled_backups.yaml`.

> 

Reviewer: It is worth putting a big WARNING here. Something like:
Please note that is not recommended to run production backups every minute. These schedules are only an example.

Author: Fixed via 5a977a20

```bash
kubectl apply -f 401_scheduled_backups.yaml
```

> After at least a minute, we should see two new pods that were created by the operator. Under the hood, these pods

Reviewer: ```suggestion
After a minute or so, we should see two new pods that were created by the operator. Under the hood, these pods
```

drwxr-xr-x 2 999 docker 220 May 17 16:39 2024-05-17.163903.zone1-2548885007
drwxr-xr-x 2 999 docker 220 May 17 16:40 2024-05-17.164003.zone1-2548885007
drwxr-xr-x 2 999 docker 220 May 17 16:41 2024-05-17.164102.zone1-2548885007
drwxr-xr-x 2 999 docker 220 May 17 16:42 2024-05-17.164202.zone1-2548885007

customer/-80/:
total 0
drwxr-xr-x 2 999 docker 220 May 17 16:38 2024-05-17.163803.zone1-0120139806
drwxr-xr-x 2 999 docker 220 May 17 16:39 2024-05-17.163902.zone1-2289928654
drwxr-xr-x 2 999 docker 220 May 17 16:40 2024-05-17.164003.zone1-2289928654
drwxr-xr-x 2 999 docker 220 May 17 16:41 2024-05-17.164102.zone1-2289928654
drwxr-xr-x 2 999 docker 220 May 17 16:42 2024-05-17.164202.zone1-2289928654

customer/80-/:
total 0
drwxr-xr-x 2 999 docker 220 May 17 16:38 2024-05-17.163812.zone1-0118374573
drwxr-xr-x 2 999 docker 220 May 17 16:39 2024-05-17.163911.zone1-4277914223
drwxr-xr-x 2 999 docker 220 May 17 16:40 2024-05-17.164010.zone1-4277914223
drwxr-xr-x 2 999 docker 220 May 17 16:41 2024-05-17.164111.zone1-4277914223
drwxr-xr-x 2 999 docker 220 May 17 16:42 2024-05-17.164208.zone1-4277914223
docker@minikube:/tmp/example$
```

### Cleanup

> Congratulations, you have correctly scheduled recurrent backups of your Vitess cluster.

Reviewer: ```suggestion
Congratulations, you have correctly scheduled recurring backups of your Vitess cluster.
```
